### PR TITLE
Add expression culture in NL2FxParameters

### PR DESCRIPTION
**Pull request by nguhoa**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/NLHandler.cs [hunk 1]
```diff
@@ -64,7 +65,13 @@ public class NL2FxParameters
...
 
         // Engine can provide ambient details for NL, such as config and feature flags.
-        public Engine Engine { get; set; }
+        public Engine Engine { get; set; }
+
+        /// <summary>
+        /// Set the locale that expect the output expression to be.
+        /// For example: "vi-VN" or "fr-FR".
+        /// </summary>
+        public CultureInfo ExpressionCultureInfo { get; set; }
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/NLHandler.cs:74]: >ExpressionCultureInfo [](http://example.com/codeflow?start=27&length=21)

nit: most places we have named CultureInfo instance as `Locale`.

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs [hunk 2]
```diff
@@ -53,7 +53,8 @@ private void PreHandleNl2Fx(IPowerFxScope scope, LanguageServerOperationContext
             {
                 Sentence = _nl2FxRequestParams.Sentence,
                 SymbolSummary = summary,
-                Engine = check.Engine
+                Engine = check.Engine,
+                ExpressionCultureInfo = check.ParserCultureInfo
```
Comment (anderson-joyle) [inline src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs:57]: What if we pass `CheckResult` only?

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/NLHandler.cs (hunk 1 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/NLHandler.cs]: update name

File: src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs (hunk 2 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs]: Do you mean when ParserCultureInfo is null? then we just have locale null and we handle it in our NL library later.

Comment (akshar2401) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs]: No! we should avoid exposing check results out of this function. This function runs behind a host (document server for example) lock and check result holds references to host's not thread safe data structures. Just get what you need from check result and pass it through this object. don't throw out the whole check result from here

Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs]: The idea was not exposing the whole object. We would pass `CheckResult` and then call something like `public Engine Engine => checkResult.Engine`. Just a suggestion. Whatever works best.

Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs]: I think Akshar suggest to not pass checkresult here so i keep this as it is to safe.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Nl2Fx/Nl2FxLanguageServerOperationHandler.cs]: yes, least privilege. 
Just culture is good here.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**